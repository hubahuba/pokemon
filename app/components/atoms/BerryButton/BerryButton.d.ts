import {BerryData} from '@/definitions/usecases/pokemon';
import {TouchableOpacityProps} from 'react-native-gesture-handler';

export interface BerryButtonProps extends TouchableOpacityProps {
  item: BerryData;
  ownedId: string;
  onPress?: (ownedId: string, item: BerryData) => void;
}
