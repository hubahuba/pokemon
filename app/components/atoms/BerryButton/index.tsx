import {View} from 'react-native';
import FastImage from 'react-native-fast-image';
import {TouchableOpacity} from 'react-native-gesture-handler';
import * as React from 'react';
import {BerryButtonProps} from '~atoms/BerryButton/BerryButton';
import {memo} from 'react';

const BerryButton = memo(function BerryButton({
  item,
  ownedId,
  onPress,
  ...others
}: BerryButtonProps) {
  return (
    <TouchableOpacity
      {...others}
      key={item.id}
      onPress={() => {
        onPress?.(ownedId, item);
      }}
      style={{height: '100%', width: '100%'}}>
      <View className="flex flex-1 p-0.5 rounded-full bg-gray-100 justify-center items-center overflow-hidden">
        <FastImage
          defaultSource={require('~/assets/images/berry.png')}
          className="w-full h-full"
          source={{uri: item.image}}
        />
      </View>
    </TouchableOpacity>
  );
});

export default BerryButton;
