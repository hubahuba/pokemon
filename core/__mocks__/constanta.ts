import {
  PokeApiQueryQuery,
  PokeGetBerriesQuery,
} from '@/definitions/gql/codegen/graphql.ts';
import {berryDataMapper, pokemonDataMapper} from '@/utils';

export const dummyData: PokeApiQueryQuery = {
  pokemon_v2_pokemon: [
    {
      id: 1,
      name: 'bulbasaur',
      weight: 69,
      pokemon_v2_pokemonsprites: [
        {
          sprites:
            'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
        },
      ],
      pokemon_v2_pokemonstats: [
        {
          base_stat: 45,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'HP',
              },
            ],
          },
        },
        {
          base_stat: 49,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Attack',
              },
            ],
          },
        },
        {
          base_stat: 49,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Defense',
              },
            ],
          },
        },
        {
          base_stat: 65,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Special Attack',
              },
            ],
          },
        },
        {
          base_stat: 65,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Special Defense',
              },
            ],
          },
        },
        {
          base_stat: 45,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Speed',
              },
            ],
          },
        },
      ],
      pokemon_v2_pokemonspecy: {
        pokemon_v2_evolutionchain: {
          pokemon_v2_pokemonspecies: [
            {
              id: 1,
              name: 'bulbasaur',
              order: 1,
              pokemon_v2_pokemons: [
                {
                  weight: 69,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 45,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 49,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 49,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 65,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 65,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 45,
                    },
                  ],
                },
              ],
            },
            {
              id: 2,
              name: 'ivysaur',
              order: 2,
              pokemon_v2_pokemons: [
                {
                  weight: 130,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 60,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 62,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 63,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 60,
                    },
                  ],
                },
              ],
            },
            {
              id: 3,
              name: 'venusaur',
              order: 3,
              pokemon_v2_pokemons: [
                {
                  weight: 1000,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 82,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 83,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                  ],
                },
                {
                  weight: 1555,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10033.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 123,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 122,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 120,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                  ],
                },
                {
                  weight: 10000,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10195.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 82,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 83,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    },
  ],
};

export const dummyNoEvo: PokeApiQueryQuery = {
  pokemon_v2_pokemon: [
    {
      id: 3,
      name: 'venusaur',
      weight: 1000,
      pokemon_v2_pokemonsprites: [
        {
          sprites:
            'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png',
        },
      ],
      pokemon_v2_pokemonstats: [
        {
          base_stat: 80,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'HP',
              },
            ],
          },
        },
        {
          base_stat: 82,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Attack',
              },
            ],
          },
        },
        {
          base_stat: 83,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Defense',
              },
            ],
          },
        },
        {
          base_stat: 100,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Special Attack',
              },
            ],
          },
        },
        {
          base_stat: 100,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Special Defense',
              },
            ],
          },
        },
        {
          base_stat: 80,
          pokemon_v2_stat: {
            pokemon_v2_statnames: [
              {
                name: 'Speed',
              },
            ],
          },
        },
      ],
      pokemon_v2_pokemonspecy: {
        pokemon_v2_evolutionchain: {
          pokemon_v2_pokemonspecies: [
            {
              id: 1,
              name: 'bulbasaur',
              order: 1,
              pokemon_v2_pokemons: [
                {
                  weight: 69,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 45,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 49,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 49,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 65,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 65,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 45,
                    },
                  ],
                },
              ],
            },
            {
              id: 2,
              name: 'ivysaur',
              order: 2,
              pokemon_v2_pokemons: [
                {
                  weight: 130,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 60,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 62,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 63,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 60,
                    },
                  ],
                },
              ],
            },
            {
              id: 3,
              name: 'venusaur',
              order: 3,
              pokemon_v2_pokemons: [
                {
                  weight: 1000,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 82,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 83,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                  ],
                },
                {
                  weight: 1555,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10033.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 123,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 122,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 120,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                  ],
                },
                {
                  weight: 10000,
                  pokemon_v2_pokemonsprites: [
                    {
                      sprites:
                        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10195.png',
                    },
                  ],
                  pokemon_v2_pokemonstats: [
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'HP',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Attack',
                          },
                        ],
                      },
                      base_stat: 82,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Defense',
                          },
                        ],
                      },
                      base_stat: 83,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Attack',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Special Defense',
                          },
                        ],
                      },
                      base_stat: 100,
                    },
                    {
                      pokemon_v2_stat: {
                        pokemon_v2_statnames: [
                          {
                            name: 'Speed',
                          },
                        ],
                      },
                      base_stat: 80,
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    },
  ],
};

export const dummyBerries: PokeGetBerriesQuery = {
  pokemon_v2_berry: [
    {
      id: 1,
      name: 'cheri',
      pokemon_v2_berryfirmness: {
        name: 'soft',
      },
      natural_gift_power: 60,
      pokemon_v2_item: {
        pokemon_v2_itemsprites: [
          {
            sprites:
              'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/items/cheri-berry.png',
          },
        ],
      },
    },
    {
      id: 2,
      name: 'chesto',
      pokemon_v2_berryfirmness: {
        name: 'super-hard',
      },
      natural_gift_power: 60,
      pokemon_v2_item: {
        pokemon_v2_itemsprites: [
          {
            sprites:
              'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/items/chesto-berry.png',
          },
        ],
      },
    },
  ],
};

export const dataPokemon = pokemonDataMapper(dummyData);
export const dataLastPokemon = pokemonDataMapper(dummyNoEvo);
export const dataBerry = berryDataMapper(dummyBerries);
