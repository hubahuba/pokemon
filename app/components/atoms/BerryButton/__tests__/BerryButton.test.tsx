import React from 'react';
import BerryButton from '../index';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {dataBerry} from '@/__mocks__/constanta.ts';

const dummyBerry = dataBerry[0];

it('renders BerryButton correctly', () => {
  const tree = render(
    <GestureHandlerRootView>
      <BerryButton item={dummyBerry} ownedId="randomString" />
    </GestureHandlerRootView>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('BerryButton action test', async () => {
  const mockSelect = jest.fn();
  const tree = render(
    <GestureHandlerRootView>
      <BerryButton
        item={dummyBerry}
        onPress={mockSelect}
        ownedId="randomString"
        testID="berryButton"
      />
    </GestureHandlerRootView>,
  );
  const {getByTestId} = tree;
  const button = getByTestId('berryButton');
  fireEvent.press(button);
  await waitFor(() => expect(mockSelect.mock.calls.length).toBe(1));
});
