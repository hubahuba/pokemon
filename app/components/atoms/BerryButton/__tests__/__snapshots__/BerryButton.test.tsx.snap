// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BerryButton correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
            {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            {
              "backgroundColor": "#f3f4f6",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "width": "100%",
                },
                {
                  "height": "100%",
                },
              ],
            ]
          }
        >
          <FastImageView
            defaultSource={
              {
                "testUri": "../../../app/assets/images/berry.png",
              }
            }
            resizeMode="cover"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
